src/nfa/nfa.cc

```diff
@@ -532,22 +532,35 @@ StateSet Nfa::get_terminating_states() const
     return revert(*this).get_reachable_states();
 }
 
-void Nfa::trim()
+void Nfa::trim(StateToStateMap* state_map)
 {
-    *this = get_trimmed_automaton();
+    if (!state_map) {
+        StateToStateMap tmp_state_map{};
+        *this = get_trimmed_automaton(&tmp_state_map);
+    } else {
+        state_map->clear();
+        *this = get_trimmed_automaton(state_map);
+    }
 }
 
-Nfa Nfa::get_trimmed_automaton() {
+Nfa Nfa::get_trimmed_automaton(StateToStateMap* state_map) {
```

Comment:
We have agreed earlier that all functions which reindex states must have state maps as output parameters. And in this particular case, since I need to know how to reindex the states in result automaton were reindexed in `reduce()` (to fill the state map in `reduce()`), I need to know how trim reindexed them as well.